src/app/merchants/approve-merchant-map.ts

```diff
@@ -0,0 +1,13 @@
+import { MerchantsRepository } from "@services/mongoose"
+
+export const approveMerchantById = async (
+  id: MerchantId,
+): Promise<BusinessMapMarker | ApplicationError> => {
+  const merchantsRepo = MerchantsRepository()
+
+  const merchant = await merchantsRepo.findById(id)
```

Comment:
I believe `findOneAndUpdate` does the full operation you're looking for in a single operation: https://mongoosejs.com/docs/tutorials/findoneandupdate.html